pkg/kgo/compression.go

```diff
@@ -274,9 +274,23 @@ func (d *decompressor) decompress(src []byte, codec byte, pool *pool.BucketedPoo
 	if compCodec == codecNone {
 		return src, nil
 	}
-	var out *bytes.Buffer
+	var (
+		out *bytes.Buffer
+		decodedBufSize int
+	)
+
+	if compCodec == codecSnappy {
+		var err error
+		decodedBufSize, err = s2.DecodedLen(src)
+		if err != nil {
+			return nil, err
+		}
+	} else {
+		// Make a guess at the output size.
+		decodedBufSize = len(src) * 2
+	}
```

Comment:
can we fold this into the `switch`? 

also lz4 has Size: `unlz4.Size()`

Reply:
_addressed @ 88058a005e32882ad66071dbf73f15eb71374cbd_